ducktape/tests/session.py

```diff
@@ -47,7 +47,12 @@ def globals(self):
         return self._globals
 
     def to_json(self):
-        return self.__dict__
+        session_context_dict = self.__dict__
+        # redact sensitive information
+        session_context_dict['_globals']['s3_access_key'] = 'xxxxx'
+        session_context_dict['_globals']['s3_region'] = 'xxxxx'
+        session_context_dict['_globals']['s3_secret_key'] = 'xxxxx'
+        return session_context_dict
```

Comment:
since these are only relevant for redpanda (they're expected in redpanda tests), it would be better if this is done in the redpanda repo.

Reply:
>it would be better if this is done in the redpanda repo.

I tried searching for places where I can modify changes but was not able to find any. We are directly supplying the results dir location and reports are produced there. To make the changes more generic, instead of directly accessing these params in session_context dic, I can check if those exists first and then override them. Let me know if you have better approach handy.

Reply:
these are not relevant in the context of upstream ducktape. also, these changes override the values, which if employed after this function returns, they'll cause the tests that rely on these values to fail. it might be better to just update the values inplace in the JSON file

Reply:
thanks! Created PR in  redpanda repo.